scripts/ci/dependency_check.bat

```diff
@@ -1,5 +1,8 @@
 pushd %~dp0..\..
 
+Rem Upgrade pip to the latest version
+python -m pip install --upgrade pip
```

Comment:
Upgrading `pip` makes `VerifyWindowsRequirements`, `VerifyLinuxRequirements` and `VerifyDarwinRequirements` have the same logic as other package building jobs, such as `BuildDebPackages`.

It can utilize the latest `pip`'s error message to better find dependency resolution failures.